Incorporating VIC 5 into the monitor workflow

hhelgason opened the pull request:
This PR includes changes to scripts that download meteorological data and run VIC. It also includes the addition of scripts that generate sub-daily forcings. Edits are made to accompanying .ecf and .cfg template files. 

To be specific, this PR includes:
1) Minor updates to README.md, ecflow_template.cfg and python_template.cfg
2) Updates to change_cfg.py to account for changes in the python configuration file. 
3) Minor edits to documentation in combine.ecf, get_metdata.ecf, regrid.ecf and reorder.ecf.
4) Edits to combine.py to account for changes in the python configuration file and other minor edits, mostly pep8 cleanup. 
5) Changes to get_metdata.py. Two variables are added (specific humidity and shortwave radiation). The script will download only one day’s worth of met data if the full year has already been downloaded.
6) subdaily_disaggregation.ecf added.
7) create_global_file.py added.

Along with other minor edits. 

This fixes #3 and resolves #4.

Commit:
Adding shortwave radiation and specific humidity to the get_metdata step. Adding subdaily disaggregation step. Solves issue #3.

Commit:
Committing Changes to the meteorological step

The get_metdata.py script is simplified, along with other minor changes.

Commit:
Committing multiple changes to the Meteorological and Run VIC steps.

Committing changes to scripts that download met data, generate subdaily forcings and run VIC, along with the accompanying ecf and configuration template files.

Commit:
Minor edits to combine.py and create_global_file.py

Commit:
Minor edits to netcdf.ecf and regrid.py

- Extra line removed from netcdf.ecf
- Unused library import deleted from regrid.py

jhamman commented on tools/bin/ecflow/processes/main/meteorological/get_metdata.py:
```diff
@@ -10,43 +10,54 @@
 old_config_file = config_dict['ECFLOW']['old_Config']
 new_config_file = config_dict['ECFLOW']['new_Config']
-N =int(config_dict['ECFLOW']['Met_Delay'])
+N = int(config_dict['ECFLOW']['Met_Delay'])
```
rename to a more useful variable name (e.g. `n_days`)

jhamman commented on tools/bin/ecflow/processes/main/meteorological/get_metdata.py:
```diff
@@ -127,68 +137,295 @@
+                        "/thredds/dodsC/MET/sph/sph_%s.nc?lon[0:1:%s]," % (lastyear, num_lon) +
+                        "lat[0:1:%s],day[%s:1:%s]," % (num_lat, lastyear_num_day, num_endofyear) +
+                        "specific_humidity[%s:1:%s]" % (lastyear_num_day, num_endofyear) +
+                        "[0:1:%s][0:1:%s]" % (num_lon, num_lat))
```
since you've just repeated nearly the same lines 12 times, I think it would make since to
1. put this into a helper function (e.g. `fill_threads_forcing_url(varname, *args)`)
2. read/fill/write these files within a for loop.

jhamman commented on tools/bin/ecflow/processes/main/meteorological/get_metdata.py:
```diff
@@ -127,68 +137,295 @@
+              "/thredds/dodsC/MET/pr/pr_%s.nc?lon[0:1:%s]," % (year, num_lon) +
+              "lat[0:1:%s],day[%s:1:%s]," % (num_lat, num_day, num_day) +
+              "precipitation_amount[%s:1:%s]" % (num_day, num_day) +
+              "[0:1:%s][0:1:%s]" % (num_lon, num_lat))
```
see comment above.

jhamman commented on tools/bin/ecflow/processes/main/meteorological/tocel_combine.py:
```diff
@@ -10,33 +10,49 @@
+else:  # if we only downloaded yesterday's data we need to append it to the existing file
+    # read in the existing file
+    existing_file = np.genfromtxt(os.path.join(
+        final_dir, '{DATA_LAT_LON}'), dtype='float')
```
this variable name is a bit misleading since `existing_file` isn't a file.

jhamman commented on tools/bin/ecflow/processes/main/models/create_global_file.py:
```diff
@@ -0,0 +1,37 @@
+                    type=argparse.FileType('r'), nargs=1,
+                    help='configuration file')
+args = parser.parse_args()
+config_dict = read_config(args.config_file[0].name)
```
I don't think you want to use the `type=argparse.FileType('r')` argument here since it automatically opens the argument and then you have to backout the filename to use it in `read_config`.

jhamman commented on tools/bin/ecflow/processes/main/models/create_global_file.py:
```diff
@@ -0,0 +1,37 @@
+# parse out year, month, and day from the model date
+model_year = model_date[:4]
+model_month = model_date[5:7]
+model_day = model_date[8:10]
```
add an example to the  comment for this section (e.g. "example 20160914")

jhamman commented on tools/bin/ecflow/processes/main/post_processing/vic2netcdf.ecf:
```diff
@@ -1,21 +0,0 @@
-%ENVIRON%/vic_utils vic2netcdf %TONIC_CFG%/VIC2nc.cfg
-
-%include <tail.h>
-
```
yay!

jhamman reviewed (changes requested):
This looks pretty good.  A few relatively minor comments.

Commit:
Committing edits following code review:

-Removed FileType from argparse and edited args.config_name call
-Simpified the get_metdata.py script with a for loop for filling in thread URLs and downloading the met data
-Added a more descriptive variable name for the met_delay variable in get_metdata.py
-A few other minor edits
